LLMonFHIR/FHIR Display/FHIRResourcesView.swift

```diff
@@ -7,17 +7,41 @@
 //
 
 import ModelsR4
+import SpeziLocalStorage
 import SpeziOpenAI
 import SwiftUI
 
-
 struct FHIRResourcesView: View {
-    @EnvironmentObject var fhirStandard: FHIR
+    @EnvironmentObject var fhirMultipleResourceInterpreter: FHIRMultipleResourceInterpreter<FHIR>
+    @EnvironmentObject var localStorage: LocalStorage<FHIR>
+
     @State var resources: [String: [FHIRResource]] = [:]
+    @State var allResourcesArray: [FHIRResource] = []
     @State var showSettings = false
+    @State var showMultipleResourcesChat = false
+    @State var interpretingMultipleResources = false
+    @State var error: String?
     @State var searchText = ""
     @AppStorage(StorageKeys.onboardingInstructions) var onboardingInstructions = true
+
+    @EnvironmentObject var fhirStandard: FHIR
+
+    var presentAlert: Binding<Bool> {
+        Binding(
+            get: {
+                error != nil
+            },
+            set: { newValue in
+                if !newValue {
+                    error = nil
+                }
+            }
+        )
+    }
     
+    private enum FHIRMultipleResourceInterpreterConstants {
+        static let storageKey = "FHIRMultipleResourceInterpreter.Cache"
+    }
```

Comment:
I updated the code so that it uses this storage key to access the saved interpretation from UserDefaults and only runs the initial interpretation if no previous interpretation was performed.